plugins/generator.md

```diff
@@ -0,0 +1,91 @@
+## Generators
+
+### Plugin generator
+
+Use the plugin generator to create a ManageIQ plugin.
+
+For example, to create a plugin called ManageIQ::MyPlugin, do the following
+
+```sh
+cd manageiq
+bundle exec rails generate manageiq:plugin ManageIQ::MyPlugin
+```
+
+This will create the directory `plugins/manageiq-my_plugin`, and you will see
+the creation progress emitted to the console:
+
+```text
+create
+create  .codeclimate.yml
+...
+```
+
+By default, the plugin will be created in the `plugins` directory, but you can
+specify an alternate directory with `--path <path>`.
+
+### Provider generator
+
+If you are creating a provider plugin, you can use the provider generator to
+emit some extra scaffolding.  The provider generator is the same as the plugin
+generator but with some extra features.
+
+For example, to create a provider plugin called ManageIQ::Providers::AwesomeCloud,
+do the following:
+
+```sh
+cd manageiq
+bundle exec rails generate manageiq:provider ManageIQ::Providers::AwesomeCloud
+```
+
+Much like the plugin generator, you can pass `--path` to change the path.
+Additionally, you also have the following options
+
+```text
+[--vcr], [--no-vcr]      # Enable VCR cassettes (Default: --no-vcr)
+[--dummy], [--no-dummy]  # Generate dummy implementations (Default: --no-dummy)
```

Comment:
What's a dummy implementation?

Reply:
It's described below in the output ... I'll tweak this to make that clearer.